gen3-integration-tests/pages/indexing_page.py

```diff
@@ -0,0 +1,100 @@
+        self.INDEX_FILE_BUTTON = "//button[contains(text(),'Index Files')]"
+        self.DOWNLOAD_LOGS_BUTTON = "//button[contains(text(), 'Download Logs')]"
+        self.FAILED_STATUS = "//div[@class='index-files-popup-text']/descendant::p[2]"
+        self.ERROR_IMAGE = "//*[name()='circle' and @id='Oval']"
+        # Locators - download manifest
+        self.DOWNLOAD_BUTTON = "//button[contains(text(),'Download')]"
+        self.SUCCESS_GREEN_MESSAGE = "//span[@class='index-files-green-label']"
+        self.DOWNLOAD_MANIFEST_BUTTON = (
+            "//button[contains(text(), 'Download Manifest')]"
+        )
+
+    def go_to(self, page: Page):
+        page.goto(self.BASE_URL)
+        screenshot(page, "IndexingPage")
+        page.wait_for_selector(self.INDEXING_PAGE, state="visible")
+        page.wait_for_selector(self.UPLOAD_MANIFEST, state="visible")
+        page.wait_for_selector(self.DOWNLOAD_MANIFEST, state="visible")
+
+    def upload_valid_indexing_manifest(self, page: Page):
+        form = page.locator(self.FILE_UPLOAD)
+        expect(form).to_be_visible
+        screenshot(page, "BeforeUploadManifest")
+        valid_manifest = (
+            TEST_DATA_PATH_OBJECT / "indexing_page" / "validTestManifest.tsv"
+        )
+        page.set_input_files(self.FILE_INPUT, valid_manifest)
+        screenshot(page, "AfterUploadManifest")
+        index_files = page.locator(self.INDEX_FILE_BUTTON)
+        index_files.click()
+        screenshot(page, "AfterClickingIndexFileButton")
+        page.wait_for_selector(self.POP_UP_BOX, state="visible")
+        page.wait_for_selector(
+            self.SUCCESS_GREEN_MESSAGE, state="visible", timeout=240000
+        )
+        page.wait_for_selector(self.DOWNLOAD_LOGS_BUTTON, state="visible")
+        page.wait_for_selector(self.DOWNLOAD_MANIFEST_BUTTON, state="visible")
+        screenshot(page, "AfterUploadIsCompleted")
+
+    def upload_invalid_indexing_manifest(self, page: Page):
+        form = page.locator(self.FILE_UPLOAD)
+        expect(form).to_be_visible
+        screenshot(page, "BeforeUploadManifest")
+        valid_manifest = (
+            TEST_DATA_PATH_OBJECT / "indexing_page" / "invalidTestManifest.tsv"
+        )
+        page.set_input_files(self.FILE_INPUT, valid_manifest)
+        screenshot(page, "AfterUploadManifest")
+        index_files = page.locator(self.INDEX_FILE_BUTTON)
+        index_files.click()
+        screenshot(page, "AfterClickingIndexFileButton")
+        page.wait_for_selector(self.POP_UP_BOX, state="visible")
+        page.wait_for_selector(self.ERROR_IMAGE, state="visible", timeout=240000)
+        failure_status = page.locator(self.FAILED_STATUS)
+        screenshot(page, "AfterFailedStatus")
+        failure_status_text = failure_status.text_content()
+        logger.info(f"Text :{failure_status_text}")
+        expect(failure_status).to_contain_text("failed")
+
+    def download_manifest(self, page: Page):
```

Comment:
this function doesn't do the actual download, so let's rename it, maybe something like `get_manifest_download_link`